mlrun/datastore/utils.py

```diff
@@ -15,21 +15,42 @@
 import tarfile
 import tempfile
 import typing
-from urllib.parse import parse_qs, urlparse
+from urllib.parse import parse_qs, urlparse, urlunparse
 
 import pandas as pd
 
 import mlrun.datastore
 
 
-def store_path_to_spark(path):
+def store_path_to_spark(path, spark_options={}):
```

Comment:
Better avoid default of dict, even though it's safe here, but it raises warnings in IDE and looks suspicious to anyone reading the code.
```suggestion
def store_path_to_spark(path, spark_options=None):
```
<img width="558" alt="Screenshot 2023-12-27 at 20 05 11" src="https://github.com/mlrun/mlrun/assets/4675524/a23786c8-0186-4af9-a994-2a701e2d2173">